internal/pkg/service/appproxy/appconfig/appconfig.go

```diff
@@ -0,0 +1,106 @@
+package appconfig
+
+import (
+	"context"
+	"net/http"
+	"time"
+
+	"github.com/benbjohnson/clock"
+	"github.com/keboola/go-client/pkg/client"
+	"github.com/keboola/go-client/pkg/request"
+	"go.opentelemetry.io/otel/attribute"
+
+	"github.com/keboola/keboola-as-code/internal/pkg/log"
+	"github.com/keboola/keboola-as-code/internal/pkg/utils/errors"
+)
+
+type Loader struct {
+	logger log.Logger
+	clock  clock.Clock
+	sender request.Sender
+	cache  map[string]cacheItem
+}
+
+type cacheItem struct {
+	config    AppProxyConfig
+	eTag      string
+	expiresAt time.Time
+}
+
+func NewLoader(logger log.Logger, clock clock.Clock, baseURL string) *Loader {
+	return &Loader{
+		logger: logger,
+		clock:  clock,
+		sender: client.New().WithBaseURL(baseURL),
+		cache:  make(map[string]cacheItem),
+	}
+}
+
+const staleCacheFallbackDuration = time.Hour
```

Comment:
Please move `const` before types, and add some comment - `staleCacheFallbackDuration is the maximum duration for which the old configuration of the application is used if loading the new one is not possible.`